src/form-builder/components/FormBuilder.jsx

```diff
@@ -85,75 +151,165 @@ export default class FormBuilder extends Component {
     this.props.saveForm(form);
   }
 
-  validateFile(file) {
+  validateExtractedZip(jsonZip) {
     const self = this;
-    const reader = new FileReader();
-    let formData = null;
+    const files = jsonZip.files;
+    const fileNames = (files || files.length < 1) && Object.keys(files);
+    if (fileNames.length < 1) {
+      self.props.onValidationError('Error Importing.. No files found in ZIP');
+      self.hideLoader();
+      return;
+    }
+    const validJsonFileNames = self.getValidJsonFileNames(fileNames);
+    const fileParsePromises = [];
+    const formJsons = [];
+    validJsonFileNames.forEach(fileName => {
+      fileParsePromises.push(jsonZip.file(fileName).async('text').then(data => {
+        try {
+          formJsons.push({ fileName, formData: JSON.parse(data) });
+        } catch (error) {
+          self.updateImportErrors(fileName, self.parseErrorMessage);
+        }
+      }));
+    });
+    Promise.all(fileParsePromises).then(() => self.processForms(formJsons));
+  }
 
+  processForms(formJsons) {
```

Comment:
how processForms different from importJsonFile() as both of them internally calling **validateFormJson** and **import** methods.. Can we extract the common code to one method and use the same in both the instances.